File: 3334-AppleRedistributionIntoBoxes/3334-AppleRedistributionIntoBoxes.js
// Last updated: 7/17/2026, 4:24:04 PM
/**
 * @param {number[]} apple
 * @param {number[]} capacity
 * @return {number}
 */
var minimumBoxes = function (apple, capacity) {
    let total = 0;

    for (let i = 0; i < apple.length; i++) {
        total = total + apple[i];
    }

    capacity.sort((a, b) => b - a);

    let boxes = 0;

    for (let i = 0; i < capacity.length; i++) {
        total = total - capacity[i];
        boxes++;

        if (total <= 0) {
            break;
        }

    }


    return boxes;
};

File: 3334-AppleRedistributionIntoBoxes/3334-AppleRedistributionIntoBoxes.cpp
// Last updated: 7/17/2026, 4:24:02 PM
class Solution {
public:
    int minimumBoxes(vector<int>& apple, vector<int>& capacity) {
        int total = 0;

        for(int i=0;i<apple.size();i++){
            total = total + apple[i];
        }

        sort(capacity.begin(),capacity.end());
        reverse(capacity.begin(),capacity.end());

        int boxes = 0;

        for(int i=0;i<capacity.size();i++){
            total = total - capacity[i];
            boxes++;

            if(total <= 0){
                break;
            }
        }

        return boxes;
    }
};